src/actions.js

```diff
@@ -55,10 +55,43 @@ export const destroyWaiter = (name) => ({
   },
 });
 
+export function waiterResponseHandler(
+  name,
+  { waiterId, responseAction, responseData }
+) {
+  return (dispatch, getState) => {
+    const waiterData = getWaiter(getState(), name);
+
+    // waiter was cleared or destroyed
+    if (waiterData.request === null) {
```

Comment:
Would this condition only happen when the waiter is cleared and _not_ destroyed? If the waiter had been destroyed, wouldn’t we get `undefined` here?